compiler/luci-interpreter/src/kernels/Transpose.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/Transpose.h"
+
+#include "kernels/Utils.h"
+
+#include <tensorflow/lite/kernels/internal/reference/reference_ops.h>
+
+#include <stdexcept>
+
+namespace luci_interpreter
+{
+
+namespace kernels
+{
+
+Transpose::Transpose(const Tensor *input, const Tensor *perm, Tensor *output)
+    : _input(input), _perm(perm), _output(output)
+{
+}
+
+void Transpose::configure()
+{
+  // Transpose op only supports 1D-5D input arrays.
+  int dims = _input->shape().num_dims();
+  const int *perm_data = getTensorData<int32_t>(_perm);
+
+  assert(_input->shape().num_dims() <= 5);
+  assert(_input->element_type() == _output->element_type());
+
+  assert(_perm->shape().num_dims() == 1);
+  assert(_perm->shape().dim(0) == dims);
+  for (int i = 0; i < dims; i++)
```

Comment:
(optional) You can fuse this loop into the below one.

Reply:
Oh! I will fix it!